crates/nargo/tests/test_data/strings/src/main.nr

```diff
@@ -1,12 +1,49 @@
-fn main(message : pub str<11>, hex_as_string : str<4>, hex_as_field : Field) {
+use dep::std;
+
+fn main(message : pub str<11>, y : Field, hex_as_string : str<4>, hex_as_field : Field) {
     let mut bad_message = "hello world";
 
     constrain message == "hello world";
     bad_message = "helld world";
+    let x = 10;
+    let z = x * 5;
+    std::println(10);
+
+    std::println(z); // x * 5 in println not yet supported
+    std::println(x);
+
+    let array = [1, 2, 3, 5, 8];
+    constrain y == 5; // Change to y != 5 to see how the later print statements are not called
+    std::println(array);
 
+    std::println(bad_message);
     constrain message != bad_message;
 
+    let hash = std::hash::pedersen([x]);
+    std::println(hash);
+    
     constrain hex_as_string == "0x41";
     // constrain hex_as_string != 0x41; This will fail with a type mismatch between str[4] and Field
     constrain hex_as_field == 0x41;
+}
+
+#[test]
+fn test_prints_strings() {
+    let message = "hello world!";
+
+    std::println(message);
+
+    let bad_message = "goodbye world";
+
+    std::println(bad_message);
+}
+
+#[test]
+fn test_prints_array() {
+    let array = [1, 2, 3, 5, 8];
+
+    std::println(array);
```

Comment:
Also curious: what does the output look like if you try to print a struct value? Do we panic?

Reply:
We panic with `thread 'main' panicked at 'print statements can only support one argument', crates/noirc_evaluator/src/ssa/acir_gen/operations/intrinsics.rs:248:5`. Good catch though as I don't really like this and will give a better error message

Reply:
I changed the panic to occur when we ssa gen the builtin rather than during acir gen. 
What is now outputted when trying to print a struct value:
```
thread 'main' panicked at 'internal error: entered unreachable code: logging structs/tuples is not supported', crates/noirc_evaluator/src/ssa/context.rs:1131:51
```